server/src/lib.rs

```diff
@@ -115,6 +115,11 @@ impl InitData {
         };
 
         let db = db::Database::new(&run.database).await?;
+
+        if run.devmode {
+            db.migrate().await?;
+        }
+
```

Comment:
Is this the answer to my question above? That it saves you a re-assignment. I can buy that.

```suggestion

        if run.devmode {
            db = db.migrate().await?;
        }

```

Reply:
Yes, that was the reason. `migrate` didn't seem to consume `self`, so I didn't want to `.clone().migrate()`, but just use `&self`.